src/core/components/navigator-stack/router.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import * as NavigatorService from 'hyperview/src/services/navigator';
+import * as Types from './types';
+import { StackNavigationState, StackRouter } from '@react-navigation/native';
+
+/**
+ * Provides a custom stack router that allows us to set the initial route
+ */
+export const Router = (stackOptions: Types.StackOptions) => {
+  const router = StackRouter(stackOptions);
+
+  /**
+   * Inject all routes into the state and set the index to the last route
+   */
+  const mutateState = (
```

Comment:
I think this function does not have any dependency with the `Router` function's scope, so it might be best pulling it out so it's not re-created every time `Router` runs, and we can later easily unit test it?